python/ray/ml/config.py

```diff
@@ -105,6 +115,47 @@ def as_placement_group_factory(self) -> "PlacementGroupFactory":
         bundles = trainer_bundle + worker_bundles
         return PlacementGroupFactory(bundles, strategy=self.placement_strategy)
 
+    @classmethod
+    def validate_config(
+        cls,
+        config: Union[ScalingConfig, "ScalingConfigDataClass"],
+        allowed_keys: Iterable[str],
+    ):
+        """
+        Validate config by raising an exception if any of allowed_keys
+        differs from the default value.
+
+        Args:
+            config: Config to check.
+            allowed_keys: ScalingConfigDataClass attribute keys that
+                can have a value different than the default one.
+        """
+        default_config = ScalingConfigDataClass()
+        if not isinstance(config, ScalingConfigDataClass):
+            config = ScalingConfigDataClass(**config)
+
+        allowed_keys = set(allowed_keys)
+
+        bad_allowed_keys = [key for key in allowed_keys if key not in config.__dict__]
+        if bad_allowed_keys:
+            raise KeyError(
+                f"Allowed key(s) {bad_allowed_keys} are not valid. "
```

Comment:
Nit: you could use a more specific word than "valid". Like, ```Allowed key(s) {bad_allowed_keys} are not `ScalingConfigDataClass` attributes.```

Reply:
We generalized this function so resolving this comment for now